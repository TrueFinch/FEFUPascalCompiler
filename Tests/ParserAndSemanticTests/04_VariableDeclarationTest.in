program a;
var
    e: record
        a: array [0.. 1000] of char;
        l: integer; 
    end;
begin
end.