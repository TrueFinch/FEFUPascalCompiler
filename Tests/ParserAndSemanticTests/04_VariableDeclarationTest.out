                         a                           
                         ┼                           
                     MainBlock                       
                ┌────────┴────────────────┐          
               var                CompoundStatement  
                ┼                                    
          SimpleVarDecl                              
 ┌──────────────┴┐                                   
 e           RecordType                              
           ┌─────┴────────┐                          
           :              :                          
    ┌──────┴─┐        ┌───┴─┐                        
    a    ArrayType    l  integer                     
          ┌──┴───┐                                   
         ..     char                                 
       ┌──┴┐                                         
       0  1000                                       